Implemented Couchbase vector store support

abhiraj2 opened the pull request:
Continues contribution from #939:
* Implements CouchbaseSearchVectorStore, including support for filter expression conversion and search with threshold.
* Add couchbase vectore store documentation
* Add spring boot autoconfig and starter
* Fixed tests that were broken in #939 

Resolves #938

sobychacko commented:
@abhiraj2 Thanks for the PR updates. I am getting a compile error when building your branch. Can you address it? Also, please DCO for committing: https://spring.io/blog/2025/01/06/hello-dco-goodbye-cla-simplifying-contributions-to-spring. You can squash all the commits into a single one since many small individual commits are in the PR. We can then add the original author as a co-author when merging.

abhiraj2 commented:
Hey @sobychacko I just checked with a fresh clone of the repo. I checkout to this PR and built it with `./mvnw clean package` as mentioned in the README and it was successfully. Can you tell me what command you are using so that I can reproduce the issue. Regarding commit squash, I'm on it.

sobychacko commented:
Ok, that sounds good then. It could be something locally I faced. Can you rebase it onto the latest main and then squash the commits? Maybe, when you squash, keep two squashes, one for your commits and the other one for the previous set of commits. If it's too much, then let's go with a single squash.

abhiraj2 commented:
Hey @sobychacko so, I rebased latest main into my branch and squashed the commits into one. But when I force push after the squash all the commits after mine are also adding me as the author.

sobychacko commented:
No worries - we will squash it when merging. Thanks!

abhiraj2 commented:
Ok @sobychacko
Thanks

sobychacko commented:
@abhiraj2 Why are we having a separate inner class for configuration? `CouchbaseSearchVectorStoreConfig`. Users have to first set everything up using `CouchbaseSearchVectorStore.CouchbaseSearchVectorStoreConfig.builder()` and then call `CouchbaseSearchVectorStore.builder(cluster, config, ...)`. Two levels of builders. How about collapsing everything into the top-level `CouchbaseSearchVectorStore.Builder`. That way, the design looks simpler. What are your thoughts?

Apart from that, I needed to make this change in one of the tests:

```
  @Bean
                public EmbeddingModel embeddingModel() {
-                       return new OpenAiEmbeddingModel(new OpenAiApi(System.getenv("OPENAI_API_KEY")));
+                       return new OpenAiEmbeddingModel(OpenAiApi.builder().apiKey(System.getenv("OPENAI_API_KEY")).build());
                }
 ```

abhiraj2 commented:
Hey @sobychacko, you are right. This is something that stayed from the previous set of commits. I'll refactor this and push changes today.

abhiraj2 commented:
Hey @sobychacko I have made the changes.

sobychacko commented:
@abhiraj2 It looks good now. One last thing before we can merge the PR - Can you sign your latest commit with the [DCO process](https://github.com/spring-projects/spring-ai/pull/2429#issuecomment-2715584119) I mentioned above? You can even add another commit with the `-s` option to avoid a force push. Then, when we merge the PR, we will squash all the commits and then add the original author as a co-author to the changes. Thanks!

sobychacko commented:
@abhiraj2 @ldoguin, This PR has been merged upstream via https://github.com/spring-projects/spring-ai/commit/d25d37ab12c0fd4eb77773d157eb15002b4ff897. Thank you both for your efforts in introducing the Couchbase vector store support in Spring AI.
